source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -26,6 +26,7 @@ DnsCacheImpl::DnsCacheImpl(
       stats_(generateDnsCacheStats(*scope_)),
       resource_manager_(*scope_, loader, config.name(), config.dns_cache_circuit_breaker()),
       refresh_interval_(PROTOBUF_GET_MS_OR_DEFAULT(config, dns_refresh_rate, 60000)),
+      timeout_interval_(PROTOBUF_GET_MS_OR_DEFAULT(config, dns_query_timeout, 5000)),
```

Comment:
I'm on the fence on whether to runtime guard this change or not. I think having a timeout here is good but it is a change. WDYT?

Reply:
I'm Ok either way - my main concern above would be if the value is OK for mobile but y'all don't have the deputy deploy problem and could just change the default if it regresses.  I think 5s should be fine for production envoy forward proxies as I doubt they're deployed places DNS takes more than 5s.

Reply:
Yeah that was my thinking. SGTM.